lib/publiccloud/provider.pm

```diff
@@ -0,0 +1,64 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Base helper class for public cloud
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+package publiccloud::provider;
+use Mojo::Base -base;
+
+has key_id     => undef;
+has key_secret => undef;
+has region     => undef;
+has prefix     => 'openqa';
+
+=head1 METHODS
+ 
+=head2 init
+ 
+Needs provider specific credentials, e.g. key_id, key_secret, region.
+
+=cut
+sub init {
+    die('init() isn\'t implemented');
```

Comment:
would say something like "Can not init base class"

Reply:
Yes, a more meaningful message makes more sense, thx.